src/main/java/org/dcsa/core/controller/BaseController.java

```diff
@@ -89,29 +94,39 @@ public void handle(Exception ex) throws Exception {
     public void handle(BadSqlGrammarException ex) {
         if ("22001".equals(ex.getR2dbcException().getSqlState())) {
             // The error with code 22001 is thrown when trying to insert a value that is too long for the column
-            if (ex.getSql().startsWith("INSERT INTO")) {
-                log.warn(this.getClass().getSimpleName() + " insert into error! - " + ex.getR2dbcException().getMessage());
-                throw new CreateException("Trying to insert a string value that is too long");
-            } else {
-                log.warn(this.getClass().getSimpleName() + " update error! - " + ex.getR2dbcException().getMessage());
-                throw new UpdateException("Trying to update a string value that is too long");
-            }
+            // - we "should" have handled that by @Size attributes, so this is mostly here as a fallback
+            throw ConcreteRequestErrorMessageException.invalidParameter("Trying to update a string value that is too long");
         } else if ("42804".equals(ex.getR2dbcException().getSqlState())) {
-            log.error(this.getClass().getSimpleName() + " database error! - " + ex.getR2dbcException().getMessage());
-            throw new DatabaseException("Internal mismatch between backEnd and database - please see log");
+            throw ConcreteRequestErrorMessageException.internalServerError("Internal mismatch between backEnd and database.", ex);
         } else {
-            log.error(this.getClass().getSimpleName() + " R2dbcException!", ex);
-            throw new DatabaseException("Internal error with database operation - please see log");
+            throw ConcreteRequestErrorMessageException.internalServerError("Internal error with database operation.", ex);
         }
     }
+
     @ExceptionHandler(ServerWebInputException.class)
     public void handle(ServerWebInputException ex) {
-        if (ex.getMessage() != null && ex.getMessage().contains("Invalid UUID string:")) {
-            log.warn("Invalid UUID string");
-            throw new InvalidParameterException("Input was not a valid UUID format");
-        } else {
-            log.error(this.getClass().getSimpleName());
-            throw ex;
+        throw ConcreteRequestErrorMessageException.invalidParameter(ex.getMessage(), ex);
+    }
+
+    @ExceptionHandler(ConcreteRequestErrorMessageException.class)
+    public ResponseEntity<RequestFailureTO> handle(HttpServletRequest httpServletRequest, ConcreteRequestErrorMessageException ex) {
```

Comment:
Use `ServerHttpRequest ` instead of `HttpServletRequest` , since we use a reactive framework this will cause issues.